cmd/kubeadm/app/util/config/common.go

```diff
@@ -48,15 +49,9 @@ func MarshalKubeadmConfigObject(obj runtime.Object) ([]byte, error) {
 	}
 }
 
-// DetectUnsupportedVersion reads YAML bytes, extracts the TypeMeta information and errors out with an user-friendly message if the API spec is too old for this kubeadm version
-func DetectUnsupportedVersion(b []byte) error {
-	gvks, err := kubeadmutil.GroupVersionKindsFromBytes(b)
-	if err != nil {
-		return err
-	}
-
-	// TODO: On our way to making the kubeadm API beta and higher, give good user output in case they use an old config file with a new kubeadm version, and
-	// tell them how to upgrade. The support matrix will look something like this now and in the future:
+// IsSupportedVersion checks if a supplied GroupVersionKind is not on the list of unsupported GVKs. If it is, an error is returned.
```

Comment:
this function does not validate `Kind`. does it make sense for it to accept GV (`GroupVersion`) instead of GVK?

i would also rename it to `IsOldSupportedGroupVersion`:
- old because it only handles old versions
- GroupVersion to indicate that it accepts GV

Reply:
+1 for the name. Thanks for that! I was feeling kind of awkward with this name.
Yes, judging from the implementation, it is more logical for it to take `GroupVersion`, but for now every call to it has to be done with `gvk.GroupVersion()` as a parameter.

Reply:
Actually the name probably has to be `IsNotOldUnsupportedVersion`. I seem to have been missing the *old* part of it.

Reply:
Yep, you are right here, but we need to be returned an error. This is a strictly utility function and it should not print directly to the user.
I can rename it to `VerifyNotOldUnsupportedVersion` and retain the error return. `Verify` prefixed funcs return errors (and not bools).

Reply:
or just name it `VerifySupportedVersion` which returns an error if it's old and unsupported?